nixos/modules/services/web-apps/akkoma.nix

```diff
@@ -0,0 +1,1031 @@
+    warnings = optionals (!config.security.sudo.enable) [''
+      The pleroma_ctl wrapper enabled by the installWrapper option relies on
+      sudo, which appears to have been disabled through security.sudo.enable.
+    ''];
+
+    users = {
+      users."${cfg.user}" = {
+        description = "Akkoma user";
+        group = cfg.group;
+        isSystemUser = true;
+      };
+      groups."${cfg.group}" = { };
+    };
+
+    # Confinement of the main service unit requires separation of the
+    # configuration generation into a separate unit to permit access to secrets
+    # residing outside of the chroot.
+    systemd.services.akkoma-config = {
+      description = "Akkoma social network configuration";
+      reloadTriggers = [ configFile ] ++ secretPaths;
+
+      unitConfig.PropagatesReloadTo = [ "akkoma.service" ];
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = true;
+        UMask = "0077";
+
+        RuntimeDirectory = "akkoma";
+        RuntimeDirectoryMode = "0711";
+
+        ExecStart =
+          (if cfg.dist.cookie == null
+            then [ genScript ]
+            else [ copyScript ])
+          ++ optionals cfg.initSecrets [ initSecretsScript ]
+          ++ [ configScript ];
+        ExecReload = optionals cfg.initSecrets [ initSecretsScript ]
+          ++ [ configScript ];
+      };
+    };
+
+    systemd.services.akkoma-initdb = mkIf cfg.initDb.enable {
+      description = "Akkoma social network database setup";
+      requires = [ "akkoma-config.service" ];
+      requiredBy = [ "akkoma.service" ];
+      after = [ "akkoma-config.service" "postgresql.service" ];
+      before = [ "akkoma.service" ];
+
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = true;
+        UMask = "0077";
+        ExecStart = initDbScript;
+        PrivateTmp = true;
+      } // optionalAttrs (db ? socket_dir || db ? socket) {
+        User = cfg.initDb.username;
+      };
+    };
+
+    systemd.services.akkoma = let
+      runtimeInputs = with pkgs; [ coreutils gawk gnused ] ++ cfg.extraPackages;
+    in {
+      description = "Akkoma social network";
+      documentation = [ "https://docs.akkoma.dev/stable/" ];
+
+      # This service depends on network-online.target and is sequenced after
+      # it because it requires access to the Internet to function properly.
+      bindsTo = [ "akkoma-config.service" ];
+      wants = [ "network-online.service" ];
+      wantedBy = [ "multi-user.target" ];
+      after = [
+        "akkoma-config.target"
+        "network.target"
+        "network-online.target"
+        "postgresql.service"
+      ];
+
+      confinement.packages = runtimeInputs;
+      path = runtimeInputs;
+
+      serviceConfig = {
+        Type = "exec";
+        User = cfg.user;
+        Group = cfg.group;
+        UMask = "0077";
+
+        RuntimeDirectory = "akkoma";
+        RuntimeDirectoryMode = "0711";
+        RuntimeDirectoryPreserve = true;
```

Comment:
Please add a comment explaining why (all the reasons, so that if they become obsolete it's obvious that really _all_ of them are obsolete).

Reply:
Will do.

Reply:
After giving it some thought, I have come to the conclusion, that the default mode of `0755` may be adequate. Using `0711` prevents listing files in the directory but still allows access to individual files if their name is known and their access mode allows for it. But as all the scripts working in it are very careful about file permissions, there is little gain in excluding read access.

I’ll still explain why the run‐time directory needs to be preserved.